src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -310,6 +325,14 @@ async ValueTask DecrementLazySolutionRefcountAsync()
                 var (solutionInfo, options) = await assetProvider.CreateSolutionInfoAndOptionsAsync(solutionChecksum, cancellationToken).ConfigureAwait(false);
                 return CreateSolutionFromInfoAndOptions(solutionInfo, options);
             }
+            catch (AssetsNotAvailableException)
+            {
+                // The work to synchronize assets and create a solution is potentially done on an entirely different
+                // thread of execution that may life on longer on the OOP side than the original N incoming calls taht
+                // caused it to be spawned.  All those incoming calls might end up cancelling, leading to the in-proc
+                // side dropping the corresponding solution.  This will appear to us as 
+                return null;
+            }
```

Comment:
note: using a new exception for this isn't great.   i'm gonig to look into how painful it would be to just detect that the sync failed and to bubble it up all the way here with a booll.